From 7a20d626883aa9d686ea2503868daec12bc636a1 Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Mon, 25 Jun 2012 15:25:53 +0200
Subject: [PATCH 38/78] wifi mac80211: fix ADDBA declined after suspend with
 wowlan

BZ: 41957

TI R5 backport
[https://github.com/TI-OpenLink/wl12xx/commit/86e32cc29a9514f5a543eaade7d7e5014472cb80]

   WLAN_STA_BLOCK_BA is set while suspending but doesn't get cleared
   when resuming in case of wowlan. This causes further ADDBA requests
   received to be rejected. Fix it by clearing it in the wowlan path
   as well.

Change-Id: I43e9b03a7351c651070792d189d474f43fc0d38e
Orig-Change-Id: Icbf6b06878000b91226255e78b498584bbf5336e
Signed-off-by: Eyal Shapira <eyal@wizery.com>
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Signed-off-by: Honore Tricot <honorex.tricot@intel.com>
Reviewed-on: http://android.intel.com:8080/69954
Reviewed-by: Haslam, AxelX <axelx.haslam@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/77995
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/net/mac80211/util.c |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/wl12xx-compat/net/mac80211/util.c b/wl12xx-compat/net/mac80211/util.c
index 98b024a..94c5f56 100644
--- a/wl12xx-compat/net/mac80211/util.c
+++ b/wl12xx-compat/net/mac80211/util.c
@@ -1319,7 +1319,13 @@ int ieee80211_reconfig(struct ieee80211_local *local)
 		}
 	}
 
-	/*
+	/* add back keys */
+	list_for_each_entry(sdata, &local->interfaces, list)
+		if (ieee80211_sdata_running(sdata))
+			ieee80211_enable_keys(sdata);
+
+ wake_up:
+    /* wifi mac80211: fix ADDBA declined after suspend with wowlan
 	 * Clear the WLAN_STA_BLOCK_BA flag so new aggregation
 	 * sessions can be established after a resume.
 	 *
@@ -1340,12 +1346,6 @@ int ieee80211_reconfig(struct ieee80211_local *local)
 		mutex_unlock(&local->sta_mtx);
 	}
 
-	/* add back keys */
-	list_for_each_entry(sdata, &local->interfaces, list)
-		if (ieee80211_sdata_running(sdata))
-			ieee80211_enable_keys(sdata);
-
- wake_up:
 	ieee80211_wake_queues_by_reason(hw,
 			IEEE80211_QUEUE_STOP_REASON_SUSPEND);
 
-- 
1.7.9

